lld/lld-c/LLDAsLibraryC.cpp

```diff
@@ -691,16 +573,63 @@ static std::string createEVMLinkerSymbolsDefinitions(
                                  numLinkerSymbols, nullptr, nullptr, 0);
 }
 
-/// Resolve undefined linker symbols in the ELF object file \inBuffer
-/// and return the obtained ELF object file.
-static bool
-resolveEVMLinkerSymbols(MemoryBufferRef inBuffer,
-                        LLVMMemoryBufferRef *outBuffer,
-                        const char *const *linkerSymbolNames,
-                        const char linkerSymbolValues[][LINKER_SYMBOL_SIZE],
-                        uint64_t numLinkerSymbols, char **errorMessage) {
+/// Returns true if the \p inBuffer contains an ELF object file.
+LLVMBool LLVMIsELFEVM(LLVMMemoryBufferRef inBuffer) {
+  Expected<ELFObjectFile<ELF32BE>> inBinaryOrErr =
+      ELFObjectFile<ELF32BE>::create(unwrap(inBuffer)->getMemBufferRef());
+  if (!inBinaryOrErr) {
+    handleAllErrors(inBinaryOrErr.takeError(), [](const ErrorInfoBase &EI) {});
+    return false;
+  }
+  return inBinaryOrErr.get().getArch() == Triple::evm;
+}
+
+/// Returns an array of undefined linker symbol names
+/// of the ELF object passed in \p inBuffer.
+void LLVMGetUndefinedReferencesEVM(LLVMMemoryBufferRef inBuffer,
+                                   char ***linkerSymbols,
+                                   uint64_t *numLinkerSymbols) {
+  if (linkerSymbols) {
```

Comment:
It doesn't seems the function does something when linkerSymbols is `nullptr`, so could we do early exit here or report fatal error?